### PR TITLE
[testfairy] Added support for folder_name and other parameters

### PR DESCRIPTION
**Pull request by gmegidish-saucelabs**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [ ] I've added or updated relevant unit tests.

### Motivation and Context

Customers of TestFairy and SauceLabs that use this plugin require passing in parameters to manage their builds. 

They need to define folder names to help with uploaded app organization. Also, apps uploaded through fastlane have open landing/download pages by default. Customers need to ability to define a closed landing page.

### Description

When uploading to TestFairy, according to it's documentation, you can upload builds using the folder_name, landing_page_mode, upload_to_saucelabs and platform parameters.

This was tested by adding these parameters to an uploading using fastlane as follows:

```
publish_to_testfairy(
    api_key: ENV["FL_TESTFAIRY_API_KEY"],
    apk: apk_file,
    folder_name: "nested/folder",
    landing_page_mode: "closed",
    upload_to_saucelabs: "on"
)
```

This yielded a successful upload with the following response:

```
{"status":"ok","build_id":"XXXX","app_name":"XXXX","app_version":"XXXX","file_size":2402770,"build_url":"XXXXX","instrumented_url":"XXXXXX","download_page_url":"https://tsfr.io/XXXX","app_url":"XXXX","invite_testers_url":"XXXX","icon_url":"https://s3.amazonaws.com/XXXX","options":"video-quality=medium,screenshot-interval=1,session-length=60m,video,logcat,shake,cpu,memory,phone-signal,battery,wifi","platform":"Android","tags":["tag1", "tag2"],"metadata":[],"has_testfairy_sdk":false,"landing_page_url":"https://tsfr.io/XXXX","saucelabs":{"app_id":"XXXX","group_id":XXXXX}}
```

### Testing Steps

You can use the code snippet above, just need a valid FL_TESTFAIRY_API_KEY and a valid .apk file.
(PS, all tests on circle-ci shows green, besides "Validate Fastlane.swift generation" which repeatedly times out after 15 minutes or so).

### CODE REVIEW

**Comment by gmegidish-saucelabs** [comment]
Who can review this please?

**Comment by renfrenkel** [comment]
I'm waiting for this feature to be merged too!

**Comment by jdgood** [comment]
This is blocking our migration off of appcenter to test fairy. Can we get someone ont he test fairy team to look at getting this merged

File: fastlane/lib/fastlane/actions/testfairy.rb
Comment (revolter) [inline fastlane/lib/fastlane/actions/testfairy.rb:270]:
```suggestion
                                       default_value: 'open',
                                       verify_block: proc do |value|
                                         UI.user_error!("The landing page mode can only be open or closed") unless %('open', 'closed').include?(value)
                                       end),
```

File: fastlane/lib/fastlane/actions/testfairy.rb
Comment (revolter) [inline fastlane/lib/fastlane/actions/testfairy.rb:275]:
```suggestion
                                       default_value: 'off',
                                       verify_block: proc do |value|
                                         UI.user_error!("The upload to Sauce Labs can only be on or off") unless %('on', 'off').include?(value)
                                       end),
```

**Review by revolter — APPROVED** [review]
LGTM!

For reference, the source of all these changes is https://docs.saucelabs.com/testfairy/api-reference/upload-api/#upload-api (expand the blue panel).

**Comment by revolter** [comment]
Would you mind adding these small fixes in this PR? 🙏🏻

**Comment by gmegidish-saucelabs** [comment]
> Would you mind adding these small fixes in this PR? 🙏🏻

@revolter done! thank you so much for the patch.  
what's the next step?

**Comment by revolter** [comment]
@gmegidish-saucelabs, thank you for applying the changes, and for all the work on this!

I pushed another commit to fix an issue with my suggestions (copied from the codebase itself), and I will merge it after the pipeline finishes.

**Pull request merged**
